libs/SalesforceSDK/src/com/salesforce/androidsdk/security/KeyStoreWrapper.java

```diff
@@ -211,6 +226,41 @@ private PrivateKey getPrivateKey(String algorithm, String name, int length) {
     }
 
     private synchronized void createKeysIfNecessary(String algorithm, String name, int length) {
+        try {
+            if (!keyStore.containsAlias(name)) {
+
+                // Generates a new key pair.
+                final KeyPairGenerator kpg = KeyPairGenerator.getInstance(algorithm, ANDROID_KEYSTORE);
+                final KeyGenParameterSpec.Builder keyGenParameterSpecBuilder = new KeyGenParameterSpec.Builder(
+                        name,
+                        KeyProperties.PURPOSE_ENCRYPT | KeyProperties.PURPOSE_DECRYPT)
+                        .setKeySize(length)
+                        .setDigests(KeyProperties.DIGEST_SHA1, KeyProperties.DIGEST_SHA256)
+                        .setEncryptionPaddings(KeyProperties.ENCRYPTION_PADDING_RSA_PKCS1, KeyProperties.ENCRYPTION_PADDING_RSA_OAEP);
+
+                /*
+                 * TODO: Remove this check once minVersion > 28.
+                 */
+                if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.P) {
+
+                    /*
+                     * Disabling StrongBox based on Google's recommendation - it's not a good
+                     * fit for this use case, since the key will need to be retrieved multiple
+                     * times. Besides, StrongBox Keymaster is available only on a few devices,
+                     * such as the Pixel 3 and Pixel 3 XL at this time.
+                     */
```

Comment:
Unrelated, but we should re-evaluate this at some point since it's been more than a few years.